plugin/src/main/java/io/snyk/eclipse/plugin/views/snyktoolview/handlers/EnableCodeProductHandler.java

```diff
@@ -0,0 +1,51 @@
+package io.snyk.eclipse.plugin.views.snyktoolview.handlers;
+
+import java.util.Map;
+
+import org.eclipse.core.commands.AbstractHandler;
+import org.eclipse.core.commands.ExecutionEvent;
+import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.jface.resource.ImageDescriptor;
+import org.eclipse.ui.PlatformUI;
+import org.eclipse.ui.commands.ICommandService;
+import org.eclipse.ui.commands.IElementUpdater;
+import org.eclipse.ui.menus.UIElement;
+
+import io.snyk.eclipse.plugin.Activator;
+import io.snyk.eclipse.plugin.properties.preferences.Preferences;
+
+public class EnableCodeProductHandler extends AbstractHandler implements IElementUpdater {
+
+	protected static ImageDescriptor IMAGE_ENABLE = Activator.getImageDescriptor("/icons/code.png");
+	protected static ImageDescriptor IMAGE_DISABLE = Activator.getImageDescriptor("/icons/code_disabled.svg");
+
+	@Override
+	public Object execute(ExecutionEvent event) throws ExecutionException {
+		String commandId = event.getCommand().getId();
+
+		ICommandService commandService = PlatformUI.getWorkbench().getService(ICommandService.class);
+		if (commandService != null) {
+			commandService.refreshElements(commandId, null);
+		}
+
+		return null;
+	}
```

Comment:
Can't this be pulled up to `AbstractHandler`?

Reply:
100%